dataset/models/tf/inception_v1.py

```diff
@@ -0,0 +1,229 @@
+                             layout=layout, name='conv', **kwargs)
+            for i in range(len(FILTERS)):
+                length = len(FILTERS[i])
+                for name, filt in zip(NAMES[:length], FILTERS[i]):
+                    net = InceptionV1.block(dim=dim, inputs=net, filters=filt, name=str(i+3)+name,\
+                                            b_norm=b_norm, **kwargs)
+                    print(name, filt, net)
+                if i != 2:
+                    net = max_pooling(dim=dim, inputs=net, **kwargs['max_pooling'])
+        return net
+
+    @staticmethod
+    def head(dim, inputs, n_outputs, head_type='dense', data_format='channels_last', is_training=True,\
+             dropout_rate=0):
+        """Head of network.
+        Consist of two kinds 'dense' and 'conv'.
+
+        Parameters
+        ----------
+        dim: int
+
+        input: tf.Tensor
+
+        n_outputs: int
+            number of outputs parameters
+
+        head_type: str {'dense', 'conv'}
+
+        data_format: str {'channels_last', 'channels_first'}
+
+        is_training: bool
+
+        dropout_rate: float
+
+        Returns
+        -------
+        net: tf.Tensor
+        """
+        with tf.variable_scope('head'):
+            if head_type == 'dense':
+                net = global_average_pooling(dim=dim, inputs=inputs, data_format=data_format)
+                if dropout_rate:
+                    net = tf.layers.dropout(net, dropout_rate, training=is_training)
+                net = tf.layers.dense(net, n_outputs)
+
+            elif head_type == 'conv':
+                net = conv_block(dim=dim, inputs=inputs, filters=n_outputs, kernel_size=1,\
+                                     layout='c', name='conv_1', data_format=data_format)
+                net = global_average_pooling(dim=dim, inputs=net, data_format=data_format)
+            else:
+                raise ValueError("Head_type should be dense or conv, but given %d" % head_type)
+        return net
+
+    def statistic(self, net, targets):
+        """Added to graph some useful funstion like accuracy or preidctions
+        Parameters:
+        -----------
+        net: tf.Tensor
+        Network output
+
+        targets: tf.Tensor
+        Answers on the data
+        """
+        prob = tf.nn.softmax(net, name='prob_predictions')
+
+        labels_hat = tf.cast(tf.argmax(prob, axis=1), tf.float32, name='predicted_labels')
+        labels = tf.cast(tf.argmax(targets, axis=1), tf.float32, name='target_labels')
+        tf.reduce_mean(tf.cast(tf.equal(labels_hat, labels), tf.float32), name='accuracy')
+
+    @staticmethod
+    def block(dim, inputs, filters, data_format='channels_last', name=None, b_norm=True, is_training=True, **kwargs):
+        """ Function contains building block from inception_v1 achitecture
+
+        Parameters:
+        -----------
+        dim: int
+        spacial dimension of input without the number of channels
```

Comment:
Docstring does not comply with numpydoc
Check examples thoroughly http://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_numpy.html